src/_pytest/runner.py

```diff
@@ -133,16 +133,14 @@ def pytest_runtest_call(item):
         pass
     try:
         item.runtest()
-    except Exception:
+    except Exception as e:
         # Store trace info to allow postmortem debugging
-        type, value, tb = sys.exc_info()
-        assert tb is not None
-        tb = tb.tb_next  # Skip *this* frame
-        sys.last_type = type
-        sys.last_value = value
-        sys.last_traceback = tb
-        del type, value, tb  # Get rid of these in this frame
-        raise
+        sys.last_type = type(e)
+        sys.last_value = e
+        # Skip *this* frame
```

Comment:
Maybe move the comment down, wrap the two statements for skipping , or leave the comment with the `= …tb_next` line?
Having it before the `assert` is a bit confusing.